### PR TITLE
kemai: init at 0.9.2

### PR DESCRIPTION
**Pull request by poelzi**
###### Description of changes

Add [kemai](https://github.com/AlexandrePTJ/kemai), a kimai desktop client.
kimai is a time tracking webapp used by companies. This client makes time entry easier.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/kemai/default.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    qt6.qtbase
+    qt6.qtconnectivity
+    qt6.qttools
+    qt6.qtlanguageserver
+    xorg.libXScrnSaver
+    magic-enum
+    spdlog
+  ];
+  cmakeFlags = [ "-DUSE_CONAN=OFF" ];
+  patches = [ ./cmake.patch ];
```
Comment (AndersonTorres) [inline pkgs/applications/misc/kemai/default.nix:34]: What this diff does?

File: pkgs/applications/misc/kemai/default.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    repo = "kemai";
+    owner = "AlexandrePTJ";
+  };
+
+  buildInputs = [
+    qt6.qtbase
+    qt6.qtconnectivity
+    qt6.qttools
+    qt6.qtlanguageserver
+    xorg.libXScrnSaver
```
Comment (AndersonTorres) [inline pkgs/applications/misc/kemai/default.nix:29]:
```suggestion
    libXScrnSaver
```

File: pkgs/applications/misc/kemai/default.nix [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  nativeBuildInputs = [ cmake qt6.wrapQtAppsHook ];
+
+  passthru = { updateScript = nix-update-script { }; };
+
+  meta = with lib; {
+    description = "Kimai desktop client written in QT6";
+    homepage = "https://github.com/AlexandrePTJ/kemai";
+    license = licenses.mit;
+    maintainers = with maintainers; [ poelzi ];
+  };
```
Comment (AndersonTorres) [inline pkgs/applications/misc/kemai/default.nix:45]: Set platforms

File: pkgs/applications/misc/kemai/default.nix [hunk 4]
```diff
@@ -0,0 +1,46 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  magic-enum,
+  spdlog,
+  qt6,
```
Comment (AndersonTorres) [inline pkgs/applications/misc/kemai/default.nix:8]: Explicitly list the qt6 components you are using.

File: pkgs/applications/misc/kemai/default.nix [hunk 5]
```diff
@@ -0,0 +1,46 @@
+{
+  lib,
```
Comment (AndersonTorres) [inline pkgs/applications/misc/kemai/default.nix:2]: Unalejandrize this.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by poelzi** [comment]
@AndersonTorres done

**Comment by AndersonTorres** [comment]
Squash all commits into one

**Comment by poelzi** [comment]
@AndersonTorres done

File: pkgs/applications/misc/kemai/default.nix (hunk 1 above)
Comment (AndersonTorres) [reply pkgs/applications/misc/kemai/default.nix:37]: Give a more descriptive name to it, something like `000-cmake-disable-conan.diff`

Also, as a matter of convention, `.patch` file extension is for mail-like commented patches (those including a header with authorship and description on the body before the diff lines), whereas `.diff` is for raw diff files.

File: pkgs/applications/misc/kemai/default.nix [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+, qttools
+, qtlanguageserver
+, wrapQtAppsHook
+, libXScrnSaver
+, nix-update-script
+}:
+
+stdenv.mkDerivation rec {
+  version = "0.9.2";
+  pname = "kemai";
```
Comment (AndersonTorres) [inline pkgs/applications/misc/kemai/default.nix:18]:
```suggestion
  pname = "kemai";
  version = "0.9.2";
```

File: pkgs/applications/misc/kemai/default.nix [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    libXScrnSaver
+    magic-enum
+    spdlog
+  ];
+  cmakeFlags = [ "-DUSE_CONAN=OFF" ];
+  patches = [ ./cmake.patch ];
+
+  nativeBuildInputs = [ cmake wrapQtAppsHook ];
+
+  passthru = { updateScript = nix-update-script { }; };
```
Comment (AndersonTorres) [inline pkgs/applications/misc/kemai/default.nix:41]:
```suggestion
  passthru = { 
    updateScript = nix-update-script { }; 
  };
```

File: pkgs/applications/misc/kemai/default.nix [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+stdenv.mkDerivation rec {
+  version = "0.9.2";
+  pname = "kemai";
+
+  src = fetchFromGitHub {
+    sha256 = "sha256-PDjNO2iMPK0J3TSHVZ/DW3W0GkdB8yNZYoTGEd2snac=";
+    rev = version;
+    repo = "kemai";
+    owner = "AlexandrePTJ";
```
Comment (AndersonTorres) [inline pkgs/applications/misc/kemai/default.nix:24]:
```suggestion
    owner = "AlexandrePTJ";
    repo = "kemai";
    rev = version;
    hash = "sha256-PDjNO2iMPK0J3TSHVZ/DW3W0GkdB8yNZYoTGEd2snac=";
```

File: pkgs/applications/misc/kemai/default.nix [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+  nativeBuildInputs = [ cmake wrapQtAppsHook ];
+
+  passthru = { updateScript = nix-update-script { }; };
+
+  meta = with lib; {
+    description = "Kimai desktop client written in QT6";
+    homepage = "https://github.com/AlexandrePTJ/kemai";
+    license = licenses.mit;
+    maintainers = with maintainers; [ poelzi ];
+    platforms   = platforms.all;
```
Comment (AndersonTorres) [inline pkgs/applications/misc/kemai/default.nix:48]:
```suggestion
    platforms   = platforms.unix;
```

Not all platforms have Qt yet.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by poelzi** [comment]
@AndersonTorres fixed

**Review by AndersonTorres — APPROVED** [review]
Good enough.

**Comment by AndersonTorres** [comment]
Ping me in 4 days to merge.

**Pull request merged**